fortuna/data/dataset/huggingface_datasets.py

```diff
@@ -0,0 +1,186 @@
+import abc
+import logging
+
+from typing import Union, Dict, List, Iterable, Optional, Tuple, Sequence
+from tqdm import tqdm
+import jax.random
+from datasets import DatasetDict, Dataset
+from jax import numpy as jnp
+from jax.random import PRNGKeyArray
+from transformers import PreTrainedTokenizer, BatchEncoding, DataCollatorWithPadding, DataCollatorForLanguageModeling
+from transformers.utils import PaddingStrategy
+
+from fortuna.data.loader.huggingface_loaders import HuggingFaceDataLoader
+from fortuna.data.loader.utils import IterableData
+from fortuna.typing import Array
+
+logger = logging.getLogger(__name__)
+
+
+class HuggingFaceClassificationDatasetABC(abc.ABC):
+    def __init__(
+        self,
+        rng: PRNGKeyArray,
+        tokenizer: PreTrainedTokenizer,
+        max_length: int,
+        padding: Union[bool, str, PaddingStrategy] = True,
+        per_device_train_batch_size: int = 8,
+        per_device_eval_batch_size: int = 8
+    ):
+        """
+        A Dataset class to work with Hugging-Face like datasets.
+
+        Parameters
+        ----------
+        rng: PRNGKeyArray
+            A PRNGKey
+        tokenizer: PreTrainedTokenizer
+            A pretrained tokenizer
+        max_length: int
+            Maximum sequence length (equal for each batch).
+            It should be se to max length accepted by the model or a smaller number.
+        padding: Union[bool, str, PaddingStrategy]
+            See `Padding and Truncation <https://huggingface.co/docs/transformers/pad_truncation>`_
+            for more information (`truncation` is always True).
+        per_device_train_batch_size: int
+            Batch size for each device to be used for training
+        per_device_eval_batch_size: int
+            Batch size for each device to be used for evaluation
+        """
+        rngs = jax.random.split(rng, 3)
+        self.rngs = {'train': rngs[0], 'validation': rngs[1], 'test': rngs[2]}
```

Comment:
In our framework, we may also want to hold out an extra data set for calibration. Would it make sense to make it appear in the split?

Reply:
I'll refactor this